rust/agama-lib/src/bootloader/http_client.rs

```diff
@@ -20,9 +20,16 @@
 
 //! Implements a client to access Agama's HTTP API related to Bootloader management.
 
-use crate::base_http_client::BaseHTTPClient;
-use crate::bootloader::model::BootloaderSettings;
-use crate::ServiceError;
+use crate::{
+    base_http_client::{BaseHTTPClient, BaseHTTPClientError},
+    bootloader::model::BootloaderSettings,
+};
+
+#[derive(Debug, thiserror::Error)]
+pub enum BootloaderHTTPClientError {
+    #[error(transparent)]
+    HTTP(#[from] BaseHTTPClientError),
```

Comment:
using transparent for base looks good to me, I would just probably use Base as enum key instead of HTTP as it is confusing with identical enum vealue from BaseHTTPClientError.

Reply:
This variant actually wraps `BaseHTTPClientError` which contains HTTP-related errors (get, post, etc.). The idea is that other errors are more domain-specific.